src/App.tsx

```diff
@@ -1,24 +1,185 @@
+    try {
+      const loadedTodos = await TodoService.get(USER_ID);
+
+      setTodos(loadedTodos);
+    } catch (err) {
+      setError(Errors.LOAD);
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return filterTodos(todos, todoStatus);
+  }, [todos, todoStatus]);
+
+  const numberOfRemainingTodos = filterTodos(todos, TodoStatus.ACTIVE).length;
+
+  const addTodo = useCallback(async (todoData: Omit<Todo, 'id'>) => {
+    setIsInputDisabled(true);
+    const temporaryTodo = {
+      id: 0,
+      ...todoData,
+    };
+
+    try {
+      setTempTodo(temporaryTodo);
+      setLoadingTodosId(currId => [...currId, temporaryTodo.id]);
+      const addedTodo = await TodoService.create(temporaryTodo);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+    } catch (err) {
+      setError(Errors.ADD);
+    } finally {
+      setTempTodo(null);
+    }
+
+    setIsInputDisabled(false);
+  }, []);
+
+  const removeTodo = useCallback(async (todoIds: number[]) => {
+    setLoadingTodosId(currId => [...currId, ...todoIds]);
+
+    try {
+      await Promise.all(todoIds.map((todoId) => TodoService.delete(todoId)));
+
+      setTodos(currTodos => currTodos.filter(
+        (todo) => !todoIds.includes(todo.id),
+      ));
+    } catch (err) {
+      setError(Errors.DELETE);
+    } finally {
+      setLoadingTodosId([]);
+    }
+  }, []);
+
+  const updateTodoData = useCallback(async (
+    todoId: number,
+    updatedTodo: Partial<Todo>,
+  ) => {
+    setLoadingTodosId((currId) => [...currId, todoId]);
+
+    try {
+      await TodoService.update(todoId, updatedTodo);
+      setTodos((currTodos) => currTodos.map(todo => (todo.id === todoId
+        ? {
+          ...todo,
+          ...updatedTodo,
+        }
+        : todo)));
+    } catch (err) {
+      setError(Errors.UPDATE);
+    } finally {
+      setLoadingTodosId(currId => currId.filter(id => id !== todoId));
+    }
+  }, []);
+
+  const allCompleted = numberOfRemainingTodos === 0;
+
+  const handleToggleAll = useCallback(async () => {
+    const notCompleted = todos.filter(({ completed }) => !completed);
+
+    if (notCompleted.length === 0) {
+      await Promise.all(
+        todos.map(todo => {
+          const updatedTodo = TodoService
+            .update(todo.id, { completed: false });
+
+          return updatedTodo;
+        }),
+      );
+
+      setTodos(prevTodos => prevTodos
+        .map(todo => ({ ...todo, completed: false })));
+    } else {
+      await Promise.all(
+        todos.map(todo => {
+          if (!todo.completed) {
+            const updatedTodo = TodoService
+              .update(todo.id, { completed: true });
+
+            return updatedTodo;
+          }
+
+          return todo;
+        }),
+      );
+
+      setTodos(prevTodos => prevTodos
+        .map(todo => ({ ...todo, completed: true })));
+    }
+  }, [todos, setTodos, TodoService.update]);
```

Comment:
Why is it needed here?) It will never be changed, because you create this class outside the component

```suggestion
  }, [todos, setTodos]);
```